test/unit/calculators/checks_benefits_support_calculator_test.rb

```diff
@@ -221,473 +335,709 @@ class CheckBenefitsSupportCalculatorTest < ActiveSupport::TestCase
+          should "be true if working over 16 hours, with children between 1 and 11" do
+            calculator = CheckBenefitsSupportCalculator.new
+            calculator.are_you_working = "yes_over_16_hours_per_week"
+            calculator.children_living_with_you = "yes"
+            %w[1_or_under 2 3_to_4 5_to_11].each do |age|
+              calculator.age_of_children = age
+              assert calculator.eligible_for_tax_free_childcare?
+            end
           end
-        end
 
-        should "return true if eligible for Tax Free Childcare with a disabled child" do
-          calculator = CheckBenefitsSupportCalculator.new
-          calculator.are_you_working = "yes_over_16_hours_per_week"
-          calculator.children_living_with_you = "yes"
-          calculator.children_with_disability = "yes"
-          %w[1_or_under 2 3_to_4 5_to_11 12_to_15 16_to_17].each do |age|
-            calculator.age_of_children = age
-            assert calculator.eligible_for_tax_free_childcare?
+          should "be true if working over 16 hours with a disabled child and children between 1 and 17" do
+            calculator = CheckBenefitsSupportCalculator.new
+            calculator.are_you_working = "yes_over_16_hours_per_week"
+            calculator.children_living_with_you = "yes"
+            calculator.children_with_disability = "yes"
+            %w[1_or_under 2 3_to_4 5_to_11 12_to_15 16_to_17].each do |age|
+              calculator.age_of_children = age
+              assert calculator.eligible_for_tax_free_childcare?
+            end
           end
         end
 
-        should "return false if not eligible for Tax Free Childcare" do
-          calculator = CheckBenefitsSupportCalculator.new
-          calculator.are_you_working = "no"
-          assert_not calculator.eligible_for_tax_free_childcare?
-
-          calculator.are_you_working = "yes"
-          calculator.children_living_with_you = "yes"
-          %w[12_to_15 16_to_17 18_to_19].each do |age|
-            calculator.age_of_children = age
-            assert_not calculator.eligible_for_tax_free_childcare?
+        context "when ineligible" do
+          should "be false if not working with children between 1 and 11" do
+            calculator = CheckBenefitsSupportCalculator.new
+            calculator.are_you_working = "no"
+            %w[1_or_under 2 3_to_4 5_to_11].each do |age|
+              calculator.age_of_children = age
+              assert_not calculator.eligible_for_tax_free_childcare?
+            end
           end
-        end
-      end
 
-      context "# eligible_for_free_childcare_2yr_olds?" do
-        should "return true if eligible for Free Childcare 2 Year Olds" do
-          calculator = CheckBenefitsSupportCalculator.new
-          %w[england wales northern-ireland].each do |country|
-            calculator.where_do_you_live = country
+          should "be false if working over 16 hours with children aged between 12 and 19" do
```

Comment:
maybe change to 'be true if working, with children' as can be working more or less than 16 hours per week and still be eligible